homeassistant/components/mqtt/siren.py

```diff
@@ -386,8 +397,10 @@ async def async_turn_off(self, **kwargs: Any) -> None:
             self._attr_is_on = False
             self.async_write_ha_state()
 
-    def _update(self, data: dict[str, Any]) -> None:
+    def _update(self, data: SirenTurnOnServiceParameters) -> None:
         """Update the extra siren state attributes."""
+        update: dict[str, Any] = dict(data)
         for attribute, support in SUPPORTED_ATTRIBUTES.items():
-            if self._supported_features & support and attribute in data:
-                self._attr_extra_state_attributes[attribute] = data[attribute]
+            assert self._attr_supported_features is not None
```

Comment:
You can drop the assert if you override it as a class attribute:
`_attr_supported_features: int`